src/inmanta/ast/__init__.py

```diff
@@ -656,9 +656,31 @@ def __init__(self, stmt: Locatable, msg: str, cause: "PluginException") -> None:
         self.__cause__: PluginException
 
     def export(self) -> export.Error:
-        error: export.Error = super().export()
-        error.category = export.ErrorCategory.plugin
-        return error
+        location: Optional[Location] = self.get_location()
+        module: Optional[str] = self.__cause__.__class__.__module__
+        name: str = self.__cause__.__class__.__qualname__
+        return export.Error(
+            type=name if module is None else "%s.%s" % (module, name),
+            message=self.__cause__.message,
+            location=location.export() if location is not None else None,
+            category=export.ErrorCategory.plugin,
+        )
+
+    def format_trace(self, indent: str = "", indent_level: int = 0) -> str:
+        """Make a representation of this exception and its causes"""
+        out = indent * indent_level + self.format()
+
+        out += "\n" + indent * indent_level + "caused by:\n"
+
+        msg_line = self.__cause__.message
+        out += (indent * (indent_level + 1)) + msg_line + "\n"
```

Comment:
Don't we want this message available in `format` as well? If so, We might be able to just keep the parent implementation for `format_trace`.

Reply:
Good suggestion. I'll do that

Reply:
done